crates/shared/src/price_estimation/instrumented.rs

```diff
@@ -58,6 +65,33 @@ impl PriceEstimating for InstrumentedPriceEstimator {
     }
 }
 
+impl<T: NativePriceEstimating> NativePriceEstimating for InstrumentedPriceEstimator<T> {
+    fn estimate_native_price(&self, token: H160) -> BoxFuture<'_, NativePriceEstimateResult> {
+        async move {
+            let start = Instant::now();
+            let estimate = self.inner.estimate_native_price(token).await;
+            self.metrics
+                .native_price_estimation_times
+                .with_label_values(&[self.name.as_str()])
+                .observe(start.elapsed().as_secs_f64());
+
+            // Count as a successful request if the answer is ok (no error) or if the error
+            // is No Liquidity
+            let success =
+                estimate.is_ok() || matches!(&estimate, Err(PriceEstimationError::NoLiquidity));
```

Comment:
I think counting `NoLiquidity` as a success makes sense but then we should also change that in the regular estimator implementation to not have a drift for no reason here.